src/rt-app.c

```diff
@@ -286,18 +286,29 @@ static int run_event(event_data_t *event, int dry_run,
 		pthread_cond_signal(&(rdata->res.cond.obj));
 		break;
 	case rtapp_barrier:
-		log_debug("barrier %s ", rdata->name);
-		pthread_mutex_lock(&(rdata->res.barrier.m_obj));
-		if (rdata->res.barrier.waiting == 0) {
-			/* everyone is already waiting, signal */
-			pthread_cond_broadcast(&(rdata->res.barrier.c_obj));
-		} else {
-			/* not everyone is waiting, mark then wait */
-			rdata->res.barrier.waiting -= 1;
-			pthread_cond_wait(&(rdata->res.barrier.c_obj), &(rdata->res.barrier.m_obj));
-			rdata->res.barrier.waiting += 1;
+		{
+			struct timespec t_delta;
```

Comment:
This seems a bit nitpicky, so feel free to ignore it, but you could put t_delta inside the if condition to avoid the additional {} indent.